### ISSUE TITLE
Installation does nothing

### ISSUE THREAD

**Issue by Genesis140** [ResetXPDR/MSFS_AutoFPS#2]
Trying to install but nothing happend, just a windows asking me if im sure to install, i agreed and ... nothing, tried several times, redownload the installer, allways same thing

**Comment by ResetXPDR** [comment]
Did you try right clicking on the installer, select properties, and check the Unblock box at the bottom of the general tab, click ok, then run the installer again per the readme?

If that didn't do it have you tried creating an exception for the installer in your AV program, or just disabled the AV program temporarily while running the installer?

Finally, did the installer file download correctly? It should be about 1.5Mb in size. If not, try redownloading again with a different browser and possibly with AV temporarily disabled if required.

**Comment by MarkB0411** [comment]
i have the same issue. Keeps saying this file is not commonly downloaded and make sure you trust it before opening. I cannot open it anyway nor change its properties.

**Comment by ResetXPDR** [comment]
@MarkB0411 Have you done the above steps?

**Comment by Genesis140** [comment]
Ok, after creating an exception in my AV all works fine

Thank you

**Comment by ResetXPDR** [comment]
I'm glad it is now working. The readme has been updated with these instructions.

**Comment by MarkB0411** [comment]
I have tried the above steps. When i right click on the installer it does not give me the properties option.

**Comment by ResetXPDR** [comment]
Can you see properties on any other file you have recently downloaded?

**Comment by ResetXPDR** [comment]
Also, try selecting the file in explorer and press ALT ENTER, which should bring up the properties window.

**Comment by ResetXPDR** [comment]
Dr Google says that if you are having issues showing properties for files, there is an issue with your windows installation. Try running SFC /SCANNOW in an admin command prompt and make sure your Windows has all the latest updates.

If it's just this file, try downloading on a different device and copying it over via USB stick.

**Comment by MarkB0411** [comment]
I can now get properties up but there is no unblock option at the bottom of the general tab.

**Comment by MarkB0411** [comment]
ah ha. All sorted i think. Thanks.

**Comment by sbrownNewsbank** [comment]
Log says:
2024-12-18 12:03:51.498 [INF] [ App:InitLog                      ] MSFS_AutoFPS v0.4.4.0 started! Log Level: Debug Log File: ..\log\MSFS_AutoFPS.log 
2024-12-18 12:03:51.499 [INF] [ App:InitSystray                  ] Creating SysTray Icon ...

**Comment by ResetXPDR** [comment]
> ah ha. All sorted i think. Thanks.

For the benefit of others experiencing this issue, what did you do to get it all sorted?

**Comment by ResetXPDR** [comment]
> After adding exception to my AV, the installer worked as expected. However, after adding an exception for the folder, the application will not open.

Does the app work if you temporarily disable your AV program? If so, that's something you need to sort out with your AV program's exclusion settings.

**Comment by sbrownNewsbank** [comment]
> > After adding exception to my AV, the installer worked as expected. However, after adding an exception for the folder, the application will not open.
> 
> Does the app work if you temporarily disable your AV program? If so, that's something you need to sort out with your AV program's exclusion settings.

No, problem remains after disabling AV all together.

**Comment by ResetXPDR** [comment]
Can you post a screenshot of your %appdata%\MSFS_AutoFPS\bin like this:
![image](https://github.com/user-attachments/assets/93e9fe1a-1879-4c47-998c-52986649cc3e)

You can directly paste a screenshot in this section.

**Comment by sbrownNewsbank** [comment]
![image](https://github.com/user-attachments/assets/08f5590d-790c-4672-b231-06fc93081db2)

**Comment by ResetXPDR** [comment]
That all looks fine. 

Are you sure the app is not starting minimised and is residing in the system tray like this:

![image](https://github.com/user-attachments/assets/1b225ccc-fa4a-444d-9792-9e86f25fdf42)

**Comment by sbrownNewsbank** [comment]
![image](https://github.com/user-attachments/assets/8ad2c4ab-dd60-49f9-856b-a8786e3e7243)

**Comment by sbrownNewsbank** [comment]
I have disabled and tried to run with no luck. I will see if I can find if has quarantined and report.

**Comment by sbrownNewsbank** [comment]
Nothing in quarantine.

**Comment by ResetXPDR** [comment]
Try reinstalling the latest test version with AV disabled and see if that works. https://github.com/ResetXPDR/MSFS_AutoFPS/releases/download/v0.4.4.1-test/MSFS_AutoFPS-installer-v0.4.4.1-test1.exe

**Comment by sbrownNewsbank** [comment]
Negative. Still nothing.
![image](https://github.com/user-attachments/assets/c899782f-44d8-4fa5-8077-dde241e14823)

![image](https://github.com/user-attachments/assets/495b1ad2-161b-4069-b428-70855f7015a6)

**Comment by ResetXPDR** [comment]
Can you send me the log file, if it got as far as creating one when you tried starting it. It's in the log directory

**Comment by sbrownNewsbank** [comment]
[MSFS_AutoFPS20241218.log](https://github.com/user-attachments/files/18189393/MSFS_AutoFPS20241218.log)

**Comment by ResetXPDR** [comment]
The log file show it is trying to start but stops after trying to create systray icon. I need to find out whether it is the icon creation stopping the app running.

Try extracting these 4 files into your bin directory and try again. Does the log file show "Icon created ..." or not?

[MSFS_AutoFPS.zip](https://github.com/user-attachments/files/18189519/MSFS_AutoFPS.zip)

**Comment by sbrownNewsbank** [comment]
Negative
2024-12-18 18:46:28.330 [INF] ----------------------------------------------------------------------- 
2024-12-18 18:46:28.342 [INF] [ App:InitLog                      ] MSFS_AutoFPS v0.4.4.1-test2 started! Log Level: Debug Log File: ..\log\MSFS_AutoFPS.log 
2024-12-18 18:46:28.343 [INF] [ App:InitSystray                  ] Creating SysTray Icon ...

**Comment by ResetXPDR** [comment]
Now we're getting somewhere. It looks like the systray icon is not being allowed to be created for some reason, throwing an exception and exiting the app.

I have expanded the code for the systray icon creation to be more descriptive of why it is failing. Try this new code:

[MSFS_AutoFPS.zip](https://github.com/user-attachments/files/18190514/MSFS_AutoFPS.zip)

Tell me if you get a message box and send the log file please.

**Comment by sbrownNewsbank** [comment]
Yes got the pop up.

Log:
2024-12-18 18:57:18.965 [INF] ----------------------------------------------------------------------- 
2024-12-18 18:57:18.977 [INF] [ App:InitLog                      ] MSFS_AutoFPS v0.4.4.1-test2 started! Log Level: Debug Log File: ..\log\MSFS_AutoFPS.log 
2024-12-18 18:57:18.978 [INF] [ App:InitSystray                  ] Creating SysTray Icon ... 
2024-12-18 18:57:18.984 [ERR] [ App:InitSystray                  ] An unexpected error occurred: The type initializer for 'H.NotifyIcon.TaskbarIcon' threw an exception. 
2024-12-18 18:57:23.033 [INF] [ App:OnExit                       ] MSFS_AutoFPS exiting ...

**Comment by ResetXPDR** [comment]
I've got to go out now but I will look in to why this is happening for you, and not the other 20,000+ users who have used this app or its predecessor apps, over the last year. Hopefully I can find a fix.

**Comment by sbrownNewsbank** [comment]
No worries. Thanks for the help!

**Comment by ResetXPDR** [comment]
One last thing to try before I go is running the app as Admin.

**Comment by sbrownNewsbank** [comment]
Running in admin give the pop up and same log as above.

**Comment by ResetXPDR** [comment]
I have a new test build that changes the way users exit the app, which gets rid of the system tray icon altogether, so this version may just fix your system's issue with icon creation. Give it a try when you can and let me know how it goes. 

https://github.com/ResetXPDR/MSFS_AutoFPS/releases/tag/v0.4.4.1-test

**Comment by ResetXPDR** [comment]
Well you've got me stumped! The app is now getting through to the point where it should should have shown the window and put an icon on the task bar, yet it fails to do so. Given so many others have this working fine, the only rational explanation for this is either something is wrong with your windows installation or there is another AV-like program blocking this app. If the former, you'd be experiencing issues with other apps, so it is probably not that, so it is most likely the latter. 

I know you have tried disabling AVG and setting exclusions but what about MS defender built in to windows, or an anti-malware program like Malwarebytes? Have you tried installing kayjay's MSFS2024_AutoFPS from here https://github.com/kayjay1c6b/MSFS2024_AutoFPS-by-kayJay_1c6/releases/tag/1.0.2.1 and does it not work with that either?

**Comment by sbrownNewsbank** [comment]
When I go to windows security settings it just links me back to my AVG. I have not tried kayjay's program.

So last night when I went to go do some flying...
I use simdashboard and Neofly...
Neofly has been giving my issues on start up. If I restart it will start up. But if I close it, I cannot re-open without a restart.
Simdashboard gave me an error last night about a simconnect config(I think it said simconnect anyway) file needing correcting and the option to let the program fix it.

Other than that, nothing weird has been happening.

**Comment by ResetXPDR** [comment]
Based on the weird experience you are having with other apps too, I think you have a mildly corrupt windows installation. Other than trying running SFC /SCANNOW in an admin command prompt and make sure your Windows has all the latest updates, there's not much else I can suggest other than a full windows reinstall, which is a pain I know but I just did it recently myself and it solved a few oddities I was experiencing with other apps.

**Comment by sbrownNewsbank** [comment]
Not sure what would cause that. Been fine for close to a year.
Anyway, thanks for your help. Hopefully I can figure it out.

**Comment by ResetXPDR** [comment]
If you don't mind @sbrownNewsbank  I'll close this as the title doesn't relate to your current issue now anyway. You can always reopen this or create a new issue if something changes on your end and me the same if I uncover something new, but for now this one has escaped me.
